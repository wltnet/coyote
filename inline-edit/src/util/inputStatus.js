export const INPUT_STATUS = {
  SAVED: "Saved",
  LOADING: "Loading",
  SUCCESS: "Success",
  ERROR: "Error",
}
