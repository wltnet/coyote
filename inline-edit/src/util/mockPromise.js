export const saveTextSuccess = (delay, value) => new Promise((resolve, reject) => {
  // 50:50 chance for resolve and reject
  const random_result = Math.random() >= 0.5 ? resolve : reject;
  setTimeout(random_result, delay, value);
});
