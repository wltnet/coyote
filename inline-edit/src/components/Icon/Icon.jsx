import React from "react";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSpinner, faCheck, faTimes } from '@fortawesome/free-solid-svg-icons';
import { INPUT_STATUS } from "../../util/inputStatus";

const Icon = ({ status }) => {
  let icon;
  let spin = false;
  switch (status) {
    case INPUT_STATUS.LOADING:
      icon = faSpinner;
      spin = true;
      break;
    case INPUT_STATUS.SUCCESS:
      icon = faCheck;
      break;
    case INPUT_STATUS.ERROR:
      icon = faTimes;
      break;
    case INPUT_STATUS.SAVED:
    default:
      icon = null;
  }

  return icon ? <FontAwesomeIcon icon={icon} spin={spin} /> : null;
}
export default Icon;
