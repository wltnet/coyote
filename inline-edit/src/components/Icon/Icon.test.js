import React from "react";
import { shallow } from "enzyme";
import Icon from "./Icon";
import { INPUT_STATUS } from "../../util/inputStatus";
// import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
// import { faSpinner, faCheck, faTimes } from '@fortawesome/free-solid-svg-icons';

jest.mock("@fortawesome/react-fontawesome", () => ({
  FontAwesomeIcon: "LaFontAwesomeIconbel",
}));

jest.mock("@fortawesome/free-solid-svg-icons", () => ({
  faSpinner: "faSpinner",
  faCheck: "faCheck",
  faTimes: "faTimes",
}));

describe("<Icon />", () => {
  it("should match the snapshot with status undefined", () => {
    const props = {
      status: undefined
    }
    const wrapper = shallow(<Icon {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should match the snapshot with status Loading", () => {
    const props = {
      status: INPUT_STATUS.LOADING,
    }
    const wrapper = shallow(<Icon {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should match the snapshot with status success", () => {
    const props = {
      status: INPUT_STATUS.SUCCESS,
    }
    const wrapper = shallow(<Icon {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should match the snapshot with status error", () => {
    const props = {
      status: INPUT_STATUS.ERROR,
    }
    const wrapper = shallow(<Icon {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
