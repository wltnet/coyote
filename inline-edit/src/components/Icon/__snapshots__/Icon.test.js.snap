// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> should match the snapshot with status Loading 1`] = `
<LaFontAwesomeIconbel
  icon="faSpinner"
  spin={true}
/>
`;

exports[`<Icon /> should match the snapshot with status error 1`] = `
<LaFontAwesomeIconbel
  icon="faTimes"
  spin={false}
/>
`;

exports[`<Icon /> should match the snapshot with status success 1`] = `
<LaFontAwesomeIconbel
  icon="faCheck"
  spin={false}
/>
`;

exports[`<Icon /> should match the snapshot with status undefined 1`] = `""`;
